test/orchard/clojuredocs_test.clj

```diff
@@ -69,54 +68,53 @@
-        (is (not (.exists cache-file)))
-        (is (empty? @docs/cache))
-        (docs/update-cache! test-edn-file)
-        (is (.exists cache-file))
-        (is (contains? @docs/cache :foo.core/bar))
-        (docs/clean-cache!)))
+      (is (not (.exists cache-file)))
+      (is (empty? @docs/cache))
+      (docs/update-cache! test-edn-file)
+      (is (.exists cache-file))
+      (is (contains? @docs/cache :foo.core/bar))
+      (docs/clean-cache!))
 
     (testing "not accessible to remote export.edn"
-      (with-redefs [docs/test-remote-url (constantly [false (IOException. "dummy")])]
-        (is (not (.exists cache-file)))
-        (is (empty? @docs/cache))
-        (is (thrown? IOException (docs/update-cache! test-edn-file)))
-        (is (not (.exists cache-file)))
-        (is (empty? @docs/cache))))))
+      (is (not (.exists cache-file)))
+      (is (empty? @docs/cache))
+      (is (thrown? Exception (docs/update-cache! "http://example.com/no/such/file.edn")))
+      (is (not (.exists cache-file)))
+      (is (empty? @docs/cache))
+      (is (thrown? Exception (docs/update-cache! "http://non.existing.server/no/such/file.edn")))
+      (is (not (.exists cache-file)))
+      (is (empty? @docs/cache)))))
 
 (deftest update-cache!-non-existing-url-test
   (let [cache-file (io/file docs/cache-file-name)]
     (is (not (.exists cache-file)))
     (is (empty? @docs/cache))
-    (is (thrown? FileNotFoundException (docs/update-cache! "file:/not/existing/file.edn")))
+    (is (thrown? Exception (docs/update-cache! "file:/not/existing/file.edn")))
     (is (not (.exists cache-file)))
     (is (empty? @docs/cache))))
 
 (deftest update-cache!-existing-cache-file-test
   (let [cache-file (io/file docs/cache-file-name)]
     (testing "no cached documentation"
-      (with-redefs [docs/test-remote-url (constantly [true])]
-        (create-dummy-cache-file now)
-        (reset! docs/cache {})
+      (create-dummy-cache-file now)
+      (reset! docs/cache {})
 
-        (is (= now (.lastModified cache-file)))
-        (is (empty? @docs/cache))
-        (docs/update-cache! test-edn-file)
-        ;; should be updated
-        (is (not= now (.lastModified cache-file)))
-        (is (contains? @docs/cache :foo.core/bar))))
+      (is (= now (.lastModified cache-file)))
+      (is (empty? @docs/cache))
+      (docs/update-cache! test-edn-file)
+      ;; should be updated
+      (is (not= now (.lastModified cache-file)))
+      (is (contains? @docs/cache :foo.core/bar)))
 
     (testing "not accessible to remote export.edn"
-      (with-redefs [docs/test-remote-url (constantly [false (IOException. "dummy")])]
-        (create-dummy-cache-file now)
-        (reset! docs/cache {})
-
-        (is (= now (.lastModified cache-file)))
-        (is (empty? @docs/cache))
-        (is (thrown? IOException (docs/update-cache! test-edn-file)))
-        (is (= now (.lastModified cache-file)))
-        (is (empty? @docs/cache))))))
+      (docs/clean-cache!)
```

Comment:
Should this be protected by the lock?

Reply:
It may be, but at the same time, it doesn't have to. Setting the cache to an empty map is not really something we care about if it happens on multiple threads simultaneously.